src/openms/source/CHEMISTRY/AASequence.cpp

```diff
@@ -843,11 +844,21 @@ namespace OpenMS
           "Cannot convert string to peptide modification: missing ')'");
     }
     ModificationsDB* mod_db = ModificationsDB::getInstance();
+    CrossLinksDB* xl_db = CrossLinksDB::getInstance();
     if (aas.peptide_.empty()) // start of peptide -> N-terminal modification?
     {
-      aas.n_term_mod_ = &(mod_db->getModification(mod, "",
-                                                  ResidueModification::N_TERM));
-      return mod_end;
+      try
+      {
+        aas.n_term_mod_ = &(mod_db->getModification(mod, "",
+                                                    ResidueModification::N_TERM));
+        return mod_end;
+      }
+      catch (...)
+      {
```

Comment:
in general we should not use Exceptions for control flow. Is there a better way to do this?